packages/salesforcedx-vscode-soql/src/editor/soqlEditorInstance.ts

```diff
@@ -164,32 +173,55 @@ export class SOQLEditorInstance {
         this.updateTextDocument(this.document, soql);
         break;
       }
+      case MessageType.UI_TELEMETRY: {
+        const { unsupported, errors } = e.payload as TelemetryModelJson;
+        if (errors) {
+          trackError('syntax_error', JSON.stringify(e.payload)).catch(
+            console.error
+          );
+        }
+        if (unsupported) {
+          // no need to duplicate.  unsupported and errors often both present
+          if (!errors) {
+            trackError('syntax_unsupported', JSON.stringify(e.payload)).catch(
+              console.error
+            );
+          }
+          channelService.appendLine(`This syntax is not yet supported.`);
+        }
+        break;
+      }
       case MessageType.SOBJECT_METADATA_REQUEST: {
-        this.retrieveSObject(e.payload as string).catch(() => {
+        this.retrieveSObject(e.payload as string).catch(async err => {
+          await showAndTrackError(MessageType.SOBJECT_METADATA_REQUEST, err);
           channelService.appendLine(
             `An error occurred while handling a request for object metadata for the ${e.payload} object.`
           );
         });
         break;
       }
       case MessageType.SOBJECTS_REQUEST: {
-        this.retrieveSObjects().catch(() => {
+        this.retrieveSObjects().catch(async err => {
+          await showAndTrackError(MessageType.SOBJECTS_REQUEST, err);
           channelService.appendLine(
             `An error occurred while handling a request for object names.`
           );
         });
         break;
       }
       case MessageType.RUN_SOQL_QUERY: {
-        this.handleRunQuery().catch(() => {
+        this.handleRunQuery().catch(async err => {
+          await showAndTrackError(MessageType.RUN_SOQL_QUERY, err);
           channelService.appendLine(
             `An error occurred while running the SOQL query.`
           );
         });
         break;
       }
       default: {
-        console.log('message type is not supported');
+        const message = `message type ${e.type} is not supported`;
```

Comment:
@ekapner - This message would be extremely rare to appear to the user in the channel output.  It's an internal error really.  There's nothing the user could do to recover.  We'd need to hot-patch to fix it.

Reply:
`Error: ${e.type}. This is an unknown error. Open an issue and supply the error message: https://github.com/forcedotcom/soql-tooling/issues/new/choose.`